pkg/tenant/api/v1/client.go

```diff
@@ -1153,3 +1166,49 @@ func (c *APIv1) SetCSRFProtect(protect bool) error {
 	c.client.Jar.SetCookies(u, cookies)
 	return nil
 }
+
+// AccessToken returns the access token cached on the client or an error if it is not
+// available. This method is primarily used for testing but can be used to fetch the
+// access token for debugging or inspection if necessary.
+func (c *APIv1) AccessToken() (_ string, err error) {
+	var cookies []*http.Cookie
+	if cookies, err = c.Cookies(); err != nil {
+		return "", err
+	}
+
+	for _, cookie := range cookies {
+		if cookie.Name == middleware.AccessTokenCookie {
+			return cookie.Value, nil
+		}
+	}
+
+	return "", ErrNoAccessToken
+}
+
+// RefreshToken returns the refresh token cached on the client or an error if it is not
+// available. This method is primarily used for testing but can be used to fetch the
+// refresh token for debugging or inspection if necessary.
+func (c *APIv1) RefreshToken() (_ string, err error) {
+	var cookies []*http.Cookie
+	if cookies, err = c.Cookies(); err != nil {
+		return "", err
+	}
+
+	for _, cookie := range cookies {
+		if cookie.Name == middleware.RefreshTokenCookie {
+			return cookie.Value, nil
+		}
+	}
+
+	return "", ErrNoRefreshToken
+}
+
+// Returns the cookies set from the previous request(s) on the client Jar.
+func (c *APIv1) Cookies() (_ []*http.Cookie, err error) {
+	if c.client.Jar == nil {
+		return nil, ErrNoCookies
+	}
+
+	cookies := c.client.Jar.Cookies(c.endpoint)
+	return cookies, nil
+}
```

Comment:
Per your question above, I did try to add access and refresh token handlers for TLS. Not quite working yet, but I made a start.